stylix/palette.nix

```
@@ -10,21 +10,49 @@ let
   opts = options.stylix;
 in
 {
+  imports = lib.singleton (
+    lib.mkRenamedOptionModule
+      [ "stylix" "polarity" ]
+      [ "stylix" "colorGeneration" "polarity" ]
+  );
   options.stylix = {
-    polarity = lib.mkOption {
-      type = lib.types.enum [
-        "either"
-        "light"
-        "dark"
-      ];
-      default = "either";
-      description = ''
-        Use this option to force a light or dark theme.
+    colorGeneration = {
+      scheme = lib.mkOption {
+        type = lib.types.enum [
+          "content"
+          "expressive"
+          "fidelity"
+          "fruit-salad"
+          "monochrome"
+          "neutral"
+          "rainbow"
+          "tonal-spot"
+          "vibrant"
+        ];
+        default = "content";
+        description = ''
+          [Matugen](https://github.com/InioX/matugen)'s color scheme type.
+        '';
+        apply = value: "scheme-${value}";
+      };
 
-        By default we will select whichever is ranked better by the genetic
-        algorithm. This aims to get good contrast between the foreground and
-        background, as well as some variety in the highlight colours.
-      '';
+      contrast = lib.mkOption {
+        type = lib.types.addCheck lib.types.float (x: x >= -1.0 && x <= 1.0);
+        default = 0.0;
+        description = ''
+          The contrast of the generated color scheme, ranging from `-1.0` to
+          `1.0` (inclusive).
+        '';
+      };
+
+      polarity = lib.mkOption {
+        type = lib.types.enum [
+          "dark"
+          "light"
+        ];
+        default = "dark";
```

Ideally, this should have an `example`:

```suggestion
        default = "dark";
        example = "light";
```